### PR TITLE
Rename `node-runtime` to `node-kitchensink-runtime`

### LINKED ISSUE #11814: Rename `node-runtime` to something containing `kitchen-sink`
the point of this is to further emphasize that `node-runtime` is NOTHING production-like, and should not be used, nor inspired from. It is, by all means, a kitchen sink:

https://stackoverflow.com/questions/33779296/what-is-exact-meaning-of-kitchen-sink-in-programming

### PR DESCRIPTION
**Pull request by wirednkod**
Rename `node-runtime` to `node-kitchensink-runtime` in order to emphasize that it is NOTHING production-like, and should not be used, nor inspired from. It is, by all means, a [kitchen sink](https://stackoverflow.com/questions/33779296/what-is-exact-meaning-of-kitchen-sink-in-programming).

Fixes #11814

### CODE REVIEW

**Comment by wirednkod** [comment]
> You can just ask for help next time. We can easily fix this together 😄 Closing and opening a new MR only gets you rid of the precious approvals.

I know :( I did not want to bother too many people and while testing a new PR I realised that all tests passed, so I figured its maybe better this way. Will keep in mind... Thank you @ggwpez 🙏🏽

File: docs/Upgrading-2.0-to-3.0.md [hunk 1]
```diff
@@ -175,9 +177,10 @@ And update the overall definition for weights on frame and a few related types a
 ##### Assets
 
 The assets pallet has seen a variety of changes:
+
 - [Features needed for reserve-backed stablecoins #7152 ](https://github.com/paritytech/substrate/pull/7152)
 - [Freeze Assets and Asset Metadata #7346 ](https://github.com/paritytech/substrate/pull/7346)
-- [Introduces account existence providers reference counting #7363 ]((https://github.com/paritytech/substrate/pull/7363))
+- [Introduces account existence providers reference counting #7363 ](<(https://github.com/paritytech/substrate/pull/7363)>)
```
Comment (ggwpez) [inline docs/Upgrading-2.0-to-3.0.md:183]: Is this intentional?

File: docs/Upgrading-2.0-to-3.0.md [hunk 2]
```diff
@@ -275,9 +278,9 @@ The pallet has been moved to a new system in which the exact amount of deposit f
  	type DesiredMembers = DesiredMembers;
  	type DesiredRunnersUp = DesiredRunnersUp;
  	type TermDuration = TermDuration;
- ```
+```
 
- **This upgrade requires storage [migration](https://github.com/paritytech/substrate/blob/master/frame/elections-phragmen/src/migrations_3_0_0.rs)**. Further details can be found in the [pallet-specific changelog](https://github.com/paritytech/substrate/blob/master/frame/elections-phragmen/CHANGELOG.md#security).
+**This upgrade requires storage [migration](https://github.com/paritytech/substrate/blob/master/frame/elections-phragmen/src/migrations_3_0_0.rs)**. Further details can be found in the [pallet-specific changelog](https://github.com/paritytech/substrate/blob/master/frame/elections-phragmen/CHANGELOG.md#security).
```
Comment (ggwpez) [inline docs/Upgrading-2.0-to-3.0.md:283]: Do you have a formatter running in your IDE or why the unrelated changes in the markdown files?

File: bin/node/runtime/Cargo.toml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "node-runtime"
+name = "kitchensink-runtime"
```
Comment (ggwpez) [inline bin/node/runtime/Cargo.toml:2]: @kianenigma Do you still want it moved to `bin/kitchensink` or not?

File: docs/Upgrading-2.0-to-3.0.md (hunk 1 above)
Comment (wirednkod) [reply docs/Upgrading-2.0-to-3.0.md:183]: No it is not 😮‍💨

File: bin/node/runtime/Cargo.toml (hunk 3 above)
Comment (wirednkod) [reply bin/node/runtime/Cargo.toml:2]: no no - Only name should change. I think structure should remain as is

File: docs/Upgrading-2.0-to-3.0.md (hunk 2 above)
Comment (wirednkod) [reply docs/Upgrading-2.0-to-3.0.md:283]: Its the formatter... I will undo and commit

File: bin/node/runtime/Cargo.toml (hunk 3 above)
Comment (wirednkod) [reply bin/node/runtime/Cargo.toml:2]: I think only name should change and not directory

**Review by ggwpez — APPROVED** [review]

File: bin/utils/subkey/src/main.rs [hunk 4]
```diff
@@ -16,7 +16,7 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! Subkey utility, based on node_runtime.
+//! Subkey utility, based on kitchensink_runtime.
```
Comment (davxy) [inline bin/utils/subkey/src/main.rs:19]:
```suggestion
//! Subkey utility
```

I don't see any link between this utility and the node/kitchensink runtime

**Review by davxy — COMMENTED** [review]
I'm pretty neutral to this renaming, but if the Substrate "example" node runtime is going to be renamed to `kitchensink` should't all the others related crates (the ones under `bin/node`) be "codenamed" `kithensink`?

E.g.  `node-executor` -> `kitchensink-executor`, `node-cli` -> `kitchensink-cli`, etc

**Comment by kianenigma** [comment]
> I'm pretty neutral to this renaming, but if the Substrate "example" node runtime is going to be renamed to `kitchensink` should't all the others related crates (the ones under `bin/node`) be "codenamed" `kithensink`?
> 
> E.g. `node-executor` -> `kitchensink-executor`, `node-cli` -> `kitchensink-cli`, etc

Good point, but not necessarily. The _node_ part is still a node, but what was previously called the _runtime of the node_ ("node-runtime") should be called kitchensink runtime.

**Review by kianenigma — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

**Comment by kianenigma** [comment]
bot merge

**Comment by kianenigma** [comment]
@alvicsam do you know what's up with CI?

**Comment by alvicsam** [comment]
yes, one more approve from our team is needed, I'll ping the team

**Review by sergejparity — APPROVED** [review]

**Comment by alvicsam** [comment]
bot merge
